core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go

```diff
@@ -152,6 +152,7 @@ type fileDetails struct {
 
 //couchDoc defines the structure for a JSON document value
 type couchDoc struct {
+	id          string
```

Comment:
We can pass `PUT db/"id" -d {_id, _rev}` to CouchDB. Hence, I do not see an issue having `id` as a string in struct and also as `_id` in the json bytes. Not a hack to me. 

If you prefer not having `id string` in the couchdoc struct, I can add the following function and use it only when the namespace is empty (but the jsonValue would be marshalled twice once at `getKey` and another and `validateAndRetrieveFields` for valid keys in an empty namespace and that's fine as we don not expect more than 3 keys for now).

```
func getKey(doc *couchDoc) (string, error)
	jsonDoc := make(jsonValue)
	decoder := json.NewDecoder(bytes.NewBuffer(doc.jsonValue))
	decoder.UseNumber()
	if err := decoder.Decode(&jsonDoc); err != nil {
		return nil, err
	}
        return jsonDoc[idField].(string)
}
```

Reply:
We need the `id` before calling the `validateAndRetrieveFields`. I will go with the `getKey()` for now.

Reply:
Not sure I understand this. Why do you need `id` before calling `validateAndRetrieveFields`. Is it not possible to just enhance the function `validateAndRetrieveFields` and get all the fields in one unmarshalling and use what is needed?

Reply:
Because `savepoint` and `channelMetadata` which are internal keys in the empty namespace would not pass the validation performed by `validateAndRetrieveFields` that is designated for the chaincode's kv pair. Hence, I introduced  `func (d *couchDoc) key() (string, error)` to get `id` field from the couchDoc.

Reply:
Ah yes,  this is a fallout of the name clash...